cli/flox/src/commands/environment.rs

```diff
@@ -1264,38 +1373,54 @@ impl Update {
             .arg("--ga-registry")
             .arg("--global-manifest")
             .arg(global_manifest_path(&flox));
-        if lockfile_path.exists() {
+        let old_lockfile = if lockfile_path.exists() {
             let canonical_lockfile_path = lockfile_path.canonicalize().map_err(|e| {
                 CoreEnvironmentError::BadLockfilePath(e, lockfile_path.to_path_buf())
             })?;
-            pkgdb_cmd.arg("--lockfile").arg(canonical_lockfile_path);
-        }
-        pkgdb_cmd.args(self.inputs);
+            pkgdb_cmd.arg("--lockfile").arg(&canonical_lockfile_path);
+            Some(serde_json::from_slice(&fs::read(canonical_lockfile_path)?)?)
+        } else {
+            None
+        };
+        pkgdb_cmd.args(self.inputs.clone());
 
         debug!("updating global lockfile with command: {pkgdb_cmd:?}");
-        let result: UpdateResult = serde_json::from_value(call_pkgdb(pkgdb_cmd)?)
+        let lockfile: Lockfile = serde_json::from_value(call_pkgdb(pkgdb_cmd)?)
             .map_err(CoreEnvironmentError::ParseUpdateOutput)?;
 
         debug!("writing lockfile to {}", lockfile_path.display());
         std::fs::write(
             lockfile_path,
-            serde_json::to_string_pretty(&result.lockfile).unwrap(),
+            serde_json::to_string_pretty(&lockfile).unwrap(),
         )
         .context("updating global inputs failed")?;
-        Ok(result.message)
+        Ok((old_lockfile, lockfile))
     }
 
-    fn update_manifest(self, flox: Flox) -> Result<String> {
-        let concrete_environment = self
-            .environment
-            .detect_concrete_environment(&flox, "update")?;
-
+    fn update_manifest(
+        &self,
+        flox: Flox,
+        concrete_environment: ConcreteEnvironment,
+    ) -> Result<UpdateResult> {
         let mut environment = concrete_environment.into_dyn_environment();
 
         environment
-            .update(&flox, self.inputs)
+            .update(&flox, self.inputs.clone())
             .context("updating environment failed")
     }
+
+    fn scrape_input(input: &FlakeRef) -> Result<()> {
+        let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
+        pkgdb_cmd
+            .args(["scrape"])
+            .arg(serde_json::to_string(&input)?)
+            // TODO: this works for nixpkgs, but it won't work for anything else
+            .arg("legacyPackages");
+
+        debug!("scraping input: {pkgdb_cmd:?}");
+        call_pkgdb(pkgdb_cmd)?;
+        Ok(())
+    }
```

Comment:
nit: should probably be on the `Lockfile` Model or somewhere else in the lib crate